### DISCUSSION TITLE
How to access parent workflow variables from a child workflow?

### DISCUSSION THREAD

**Discussion by ricksearcy** [elsa-workflows/elsa-core#5234] [Design & tech]
We have a situation where we need to pass data from a parent workflow to a child workflow that is dispatched from a dispatch workflow activity. Once the child workflow activity is executed we can not access the Variables of the parent workflow. We can only access the Variables of the executing child workflow. This is an issue because we lose context information about the Root parent workflow.

**Comment by glime-ai** [comment]
To pass data from a parent workflow to a child workflow in Elsa, you can use the `Input` property of the `DispatchWorkflow` activity to send the necessary data to the child workflow. When you configure the `DispatchWorkflow` activity in the parent workflow, you can specify the input data that the child workflow should receive.

Here's how you can do it:

1. In the parent workflow, when you set up the `DispatchWorkflow` activity, you can define the input that the child workflow will receive. You can pass the `workflowInstanceId` of the parent workflow as part of this input.

2. In the child workflow, you can access the input data that was passed from the parent workflow. If you need to access the `workflowInstanceId` of the parent workflow, you would retrieve it from the input that was passed to the child workflow.

For example, if you're using the designer, you can set the `Input` property of the `DispatchWorkflow` activity using a JavaScript expression like this:

```javascript
return {"ParentWorkflowInstanceId": workflowExecutionContext.workflowInstance.id};
```

Then, in the child workflow, you can access this input using an activity that can read input variables, or if you're writing custom activities, you can access the input through the `ActivityExecutionContext` object.

Please note that the child workflow will have its own `workflowInstanceId`, which is different from the parent workflow's `workflowInstanceId`. If you need to maintain a reference to the parent workflow's ID within the child workflow, you must explicitly pass it as shown above.

If you're working with custom activities and need to access the parent workflow's `workflowInstanceId` from within the child workflow's code, you would typically access the input that was passed to the child workflow through the `ActivityExecutionContext` object.

If you need further clarification or assistance with the specific code or configuration, please provide more details about your current setup and where you're encountering difficulties.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by ricksearcy** [reply]
I need to be able to set an Output on the server in C# and then be able you get that Output in a child workflow without using the designer.
